### PR TITLE
HDEUS-936 refactor testing framework

### PR DESCRIPTION
**Pull request by amin010**
The old framework methods are untouched so current tests won't break

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -10,6 +10,8 @@ logger = ">=1.4"
 sparkpipelineframework = ">=0.1.99"
 sparkdataframecomparer = ">=0.1.68"
 Deprecated = "==1.2.12"
+requests = ">=2.25.1"
+dictdiffer = ">=0.8.1"
```
Comment (imranq2) [inline Pipfile:14]: should these be in dev-packages?

File: docker-compose.yml [hunk 2]
```diff
@@ -9,3 +9,53 @@ services:
       - ./:/spftest/
     container_name: spftest_dev
     working_dir: /spftest
+  mock-server:
```
Comment (imranq2) [inline docker-compose.yml:12]: we had this in the docker-services to reduce the containers run by default.  is there a reason we moved it?

**Review by imranq2 — APPROVED** [review]

File: docker-compose.yml (hunk 2 above)
Comment (amin010) [reply docker-compose.yml:12]: @imranq2 sorry not sure what you mean. This project (SAM.Testing) has one docker-compose file (`docker-compose.yml`) to test the new testing framework we need new docker services e.g. mock-server. maybe you're talking about helix.pipeline project?

File: Pipfile (hunk 1 above)
Comment (amin010) [reply Pipfile:14]: sure.

Comment (amin010) [reply Pipfile:14]: sure

**Pull request merged**